### ISSUE TITLE
End to end model.fit() performance testing w/ COCO metrics

### ISSUE THREAD

**Issue by LukeWood** [keras-team/keras-cv#16]
Some comments on this:
- we can use a Keras callback to track the runtimes of each step
- we must omit the first batch, as this will include the Keras compilation step
- we can test the model runtime with/without each metric, this will give us a reference as to how expensive the metrics are.  No metric should increase runtime by more than a factor of 2x

**Comment by bhack** [comment]
Can we reuse https://github.com/tensorflow/benchmarks/tree/master/perfzero ?
